packages/just-scripts-utils/src/__tests__/findMonoRepoRootPath.spec.ts

```diff
@@ -0,0 +1,77 @@
+import path from 'path';
+import mockfs from 'mock-fs';
+import { paths } from '../paths';
+import { findMonoRepoRootPath } from '../findMonoRepoRootPath';
+
+// TODO: make sure tests pass on windows
+
+describe('findMonoRepoRootPath', () => {
+  const packageJsonWithJustStack = JSON.stringify({ just: { stack: 'just-stack-monorepo' } });
+  const packageJsonWithWrongStack = JSON.stringify({ just: { stack: 'no' } });
+
+  afterEach(() => {
+    paths.projectPath = '';
+    mockfs.restore();
+  });
+
+  it('works when there is no monorepo root', () => {
+    mockfs({
+      '/a': {},
+      '/b/c/d': {},
+      '/e': {
+        'package.json': '{}', // no just.stack
+        f: {
+          'package.json': packageJsonWithWrongStack,
+          g: {}
+        }
+      },
+      '/h': {
+        i: {}, // this will be the projectPath
+        j: { 'rush.json': '{}' } // rush.json is in the wrong place
+      }
+    });
+
+    paths.projectPath = '/a';
+    expect(findMonoRepoRootPath()).toBeNull();
+
+    paths.projectPath = '/b/c/d';
+    expect(findMonoRepoRootPath()).toBeNull();
+
+    paths.projectPath = '/e/f/g';
+    expect(findMonoRepoRootPath()).toBeNull();
+
+    paths.projectPath = '/h/i';
+    expect(findMonoRepoRootPath()).toBeNull();
+  });
+
+  it('works when projectPath is the monorepo root', () => {
+    mockfs({
+      '/a': { 'rush.json': '{}' },
+      '/b': { 'package.json': packageJsonWithJustStack }
+    });
+
+    paths.projectPath = '/a';
+    expect(findMonoRepoRootPath()).toBe('/a');
+
+    paths.projectPath = '/b';
```

Comment:
I think the tests should be split into these:

```
it('finds monorepo root path when there is a rush.json at root', () => {...});
it('finds monorepo root path when there is a package.json with a just.stack key', () => {...});
```

Reply:
This just makes tests to be more focused so when it fails, it's clearer which case was busted.

Reply:
In theory I agree, but main reason I combined them is that I'm worried about the cost of setting up and tearing down the mock filesystem each time (haven't measured it so not sure how much of an actual issue that is). Jest is pretty good about telling you exactly which line failed, which makes multiple cases in a single test less problematic practically.

Reply:
I kind of want to leave these two together because I can't figure out good descriptions for them separately... :P The point of both tests is to verify that the utility works when `projectPath` *is also the monorepo root*.